### PR TITLE
Add SPI and App components supporting `TransactionDispatcher` for modularized HCS

### LINKED ISSUE #4960: Install workflow-aware `TransitionRunner` module to `HederaApp`
[Here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/TransitionRunner.java#L107), the `TransitionRunner` extracts the type of the operation being handled.

In `mono-service`, the next step is to look up a `TransitionLogic` implementation.

In `hedera-app`, we want to replace this with a call to a `Handler` method.

To satisfy #4945, it is not necessary to replace everything outside `TransitionRunner`; but we _do_ need to build `HederaApp` with a different binding for `TransitionRunner`. 

Currently `TransitionRunner` is an "implicit binding" (i.e. has an `@Inject`-labeled constructor that Dagger can use). We need to instead make it an explicit binding that we can control, and give `HederaApp` the workflow-aware implementation.

- [x] Define an interface for concrete `TransitionRunner` so components that depend on it can accept either the workflow-aware or legacy implementation.
- [x] Write a workflow-aware implementation under `hedera-app` (this will be largely duplicated; though not entirely); check the `staticProperties.workflowsEnabled()` property to check if the operation is in this `Set<HederaFunctionality>`.
- [x] Update `HederaApp` to be built with a binding of the workflow-aware implementation (possibly provided by the same `TransactionsModule` mentioned below?)
- [x] To provide access to `ConsensusCreateTopicHandler`, bind a `TransactionDispatcher` inside `HederaApp` by adding e.g. a `TransactionsModule` installed by `HederaApp` that provides the `TransactionHandlers` dependency a la [here](https://github.com/hashgraph/hedera-services/blob/mhs-adapter/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryModule.java#L63); inject the dispatcher into the workflow-aware `TransitionRunner`.

### LINKED ISSUE #4961: Update `HederaMerkleState.handleConsensusRound()` to be workflow-aware (ok to duplicate code from `ServicesState.handleConsensusRound()`).

### LINKED ISSUE #4962: Define minimal record-builder API to capture the information that HCS `TransitionLogic`'s put in the receipt (`TopicID` of created topic for `CreateTopic` and running hash for `ConsensusSubmitMessage`)---implement via `mono-service` adapter.

### LINKED ISSUE #4963: Use Config Facility Adaptor (c.f. PR #4904) to provide config needed; e.g. so `TopicCreate` can reject creation of topic once the configured system limit is reached---implement via `mono-service` adapter. (Or not?)

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Updates `TransactionDispatcher` to dispatch functions in `hedera.workflows.enabled` to appropriate `TransactionHandler` implementations.
 - Supports dispatch to HCS transactions with minimal required `ConsensusConfig`, `HandleContext`, and `RecordBuilder` interfaces.
     * `HandleContext` most notably includes an `EntityExpiryValidator` to allow `CreateTopic` and `UpdateTopic` to enforce the standard auto-renew/expiry policy.
     * `RecordBuilder` has an abstract base class `UniversalRecordBuilder` from which to derive transaction-specific builders; each `TransactionHandler` must provide an appropriate instance via a `newRecordBuilder()` override.
- Closes #4960 
- Closes #4961
- Closes #4962
- Closes #4963

### CODE REVIEW

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/exceptions/HandleStatusException.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.exceptions;
+
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+
+/**
+ * A runtime exception that wraps a {@link ResponseCodeEnum} status.
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/exceptions/HandleStatusException.java:22]: We should probably add when this exception will be thrown ?

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/validation/ExpiryMeta.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        return autoRenewPeriod != NA;
+    }
+
+    public boolean hasAutoRenewNum() {
+        return autoRenewNum != NA;
+    }
+
+    public boolean hasFullAutoRenewSpec() {
+        return hasAutoRenewNum() && hasAutoRenewPeriod();
+    }
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/validation/ExpiryMeta.java:49]: Need javadocs on these methods 😀

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/meta/MonoHandleContext.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.Key;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Instant;
+import java.util.Objects;
+import java.util.function.LongSupplier;
+import javax.inject.Singleton;
+
+/**
+ * A {@link HandleContext} implementation that primarily uses adapters of {@code mono-service}
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/meta/MonoHandleContext.java:38]: We should mention all the adapters need to be re-written or classes from mono-service should be ported to `hedera-app` in future.

**Review by qnswirlds — DISMISSED** [review]
Approve for gradle and module-info changes.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/validation/MonoExpiryValidator.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    private boolean hasCompleteAutoRenewSpec(
+            final boolean entityCanSelfFundRenewal, final ExpiryMeta creationMetadata) {
+        return creationMetadata.hasFullAutoRenewSpec()
+                || (!creationMetadata.hasExplicitExpiry() && entityCanSelfFundRenewal);
+    }
+
+    private void validateAutoRenewAccount(final long autoRenewNum) {
+        final var autoRenewId = new Id(numbers.shard(), numbers.realm(), autoRenewNum);
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/validation/MonoExpiryValidator.java:136]: Need javadocs on these methods

**Review by Neeharika-Sompalli — DISMISSED** [review]
Looks Great ! Thanks @tinker-michaelj . Most comments are for missing javadocs and are not blocking this PR.

**Review by kimbor — DISMISSED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Thanks @tinker-michaelj

**Review by kimbor — APPROVED** [review]

**Pull request merged**